lib/normalize_grammar.re

```diff
@@ -76,7 +76,21 @@ and normalize_body = (rule_body) => {
       let xs = List.map(normalize_to_simple, bodies);
```

Comment:
you should check for the presence of BLANK here, on bodies, before the normalize_to_simple calls.
BLANK is always the last element of CHOICE, so you can do
| A.CHOICE(bodies) => {
  switch (List.rev(bodies)) {
 | [BLANK] -> failwith("Impossible, a single BLANK in a CHOICE")
 | [BLANK, other_body] ->
    let (simple, rest) = normalize_to_simple(other_body);
    (B.OPTION(simple), rest)
| [BLANK,  ...other_bodies] ->
  /* this will create a gensymed ATOM(SYMBOL) for this CHOICE */
  let (simple, rest) = normalize_to_simple(A.CHOICE(other_bodies));
  (B.OPTION(simple), rest)

Reply:
Ok addressed